PWGJE/Tasks/hadronPhotonCorrelation.cxx

```diff
@@ -128,55 +159,179 @@ struct HadronPhotonCorrelation {
-    registry.add("reconstructed/hadrons/hHadronPtPrimReco", "Reconstructed Primaries Spectrum", kTH1F, {axisPtAssoc});  // Primary hadron spectrum
+    registry.get<THnSparse>(HIST("reconstructed/hadrons/hHadronSpectrumReco"))->Sumw2();
+    registry.add("reconstructed/hadrons/hHadronPtPrimReco", "Reconstructed Primaries Spectrum", kTH1F, {axisPtAssoc}); // Primary hadron spectrum
+    registry.get<TH1>(HIST("reconstructed/hadrons/hHadronPtPrimReco"))->Sumw2();
     registry.add("reconstructed/hadrons/hHadronPtSecReco", "Reconstructed Secondaries Spectrum", kTH1F, {axisPtAssoc}); // Secondary hadron spectrum
+    registry.get<TH1>(HIST("reconstructed/hadrons/hHadronPtSecReco"))->Sumw2();
 
     // Photons
     registry.add("reconstructed/photons/hPhotonCorrelReco", "Reconstructed Trigger-Photon Correlation", kTHnSparseF, {axisPtTrig, axisPtAssoc, axisDeltaEta, axisDeltaPhi});
+    registry.get<THnSparse>(HIST("reconstructed/photons/hPhotonCorrelReco"))->Sumw2();
     registry.add("reconstructed/photons/hPhotonMultReco", "Reconstructed Photon Multiplicity", kTH1F, {axisMult});
+    registry.get<TH1>(HIST("reconstructed/photons/hPhotonMultReco"))->Sumw2();
     registry.add("reconstructed/photons/hPhotonSpectrumReco", "Reconstructed Photon Spectrum", kTHnSparseF, {axisPtAssoc, axisEtaAssoc, axisPhi});
+    registry.get<THnSparse>(HIST("reconstructed/photons/hPhotonSpectrumReco"))->Sumw2();
 
     // Charged Pions
     registry.add("reconstructed/charged/hPionCorrelReco", "Reconstructed Trigger-Pion Correlation", kTHnSparseF, {axisPtTrig, axisPtAssoc, axisDeltaEta, axisDeltaPhi});
+    registry.get<THnSparse>(HIST("reconstructed/charged/hPionCorrelReco"))->Sumw2();
     registry.add("reconstructed/charged/hPionMultReco", "Reconstructed Pion Multiplicity", kTH1F, {axisMult});
+    registry.get<TH1>(HIST("reconstructed/charged/hPionMultReco"))->Sumw2();
     registry.add("reconstructed/charged/hPionSpectrumReco", "Reconstructed Pion Spectrum", kTHnSparseF, {axisPtAssoc, axisEtaAssoc, axisPhi});
+    registry.get<THnSparse>(HIST("reconstructed/charged/hPionSpectrumReco"))->Sumw2();
+  }
+
+  // Get pTHat of the event
+  template <typename P>
+  double getCollisionPtHat(const P& particles)
+  {
+    double pTHat = 0;
+    bool isDiffractive = false;
+
+    for (const auto& particle : particles) {
+
+      if (std::abs(particle.getGenStatusCode()) == pythiaCodeOutgoingHard && !isDiffractive) {
+        if (particle.pt() > pTHat) {
+          pTHat = particle.pt();
+        }
+      }
+      if (std::abs(particle.getGenStatusCode()) == pythiaCodeOutgoingDiff) {
+        if (isDiffractive) {
+          if (particle.pt() > pTHat) {
+            pTHat = particle.pt();
+          }
+        } else {
+          pTHat = particle.pt();
+          isDiffractive = true;
+        }
+      }
+    }
+
+    return pTHat;
+  }
+
+  template <typename P>
+  bool rejectOutliers(const P& particles, double pTHat)
```

Comment:
do we have central functions for this in JE?

Reply:
SImilar to the pTHat, in the [fullJetSpectra](https://github.com/AliceO2Group/O2Physics/blob/44edb11daa411d2a53d4dba7bfc1ed97226477c9/PWGJE/Tasks/fullJetSpectra.cxx#L1165) task, outlier rejection is done explicitly